pkg/bundle/release.go

```diff
@@ -6,64 +6,61 @@ import (
 	"context"
 	"crypto/tls"
 	"crypto/x509"
-	"encoding/json"
 	"errors"
 	"fmt"
-	"io/ioutil"
 	"net/http"
 	"net/url"
 	"os"
 	"path/filepath"
+	"runtime"
 	"strings"
 
 	semver "github.com/blang/semver/v4"
 	"github.com/google/uuid"
 	"github.com/openshift/oc/pkg/cli/admin/release"
 	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
 
 	"github.com/RedHatGov/bundle/pkg/cli"
 	"github.com/RedHatGov/bundle/pkg/config"
 	"github.com/RedHatGov/bundle/pkg/config/v1alpha1"
 	"github.com/RedHatGov/bundle/pkg/image"
 )
 
-// This file is for managing OCP release related tasks
-
-// import(
-//   "github.com/openshift/cluster-version-operator/pkg/cincinnati"
-// )
+// archMap maps Go architecture strings to OpenShift supported values for any that differ.
+var archMap = map[string]string{
+	"amd64": "x86_64",
+}
 
 // ReleaseOptions configures either a Full or Diff mirror operation
 // on a particular release image.
 type ReleaseOptions struct {
 	cli.RootOptions
 	release string
+	arch    string
 }
 
 // NewReleaseOptions defaults ReleaseOptions.
-func NewReleaseOptions(ro cli.RootOptions) *ReleaseOptions {
-	return &ReleaseOptions{RootOptions: ro}
-}
+func NewReleaseOptions(ro cli.RootOptions, flags *pflag.FlagSet) *ReleaseOptions {
+	var arch string
+	opts := ro.FilterOptions
+	opts.Complete(flags)
+	if opts.IsWildcardFilter() {
+		arch = runtime.GOARCH
```

Comment:
This behavior is a little weird. If the user wants to filter no platforms, then the local platform is used, which is not what was asked for. Is there a better way to use a wildcard to determine arch, like query `api.openshift.com` for available platforms or hardcode a known list of platforms?

Reply:
@estroz Very good point here. I can totally see where you are coming from. I was kind of trying to recreate the options use on `oc release`, but I can see how that I was toying with the idea of of using the `oc image info` or `oc release info` library to gather that info from the API response.

Reply:
If arch is omitted from the api query, the multi-arch images are returned

Reply:
Then for now I can make that query parameter to the API server optional, but that doesn't really solve the issue for @estroz's other comment with the example `linux/arm/*`. We probably should have knowledge of what is support so we can reported back invalid inputs as well.